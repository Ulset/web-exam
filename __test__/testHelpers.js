const userData = {id: 3, firstname: "Sander", lastname: "Ulset", email: "sander.ulset@gmail.com"}
const userToken = "Bearer testTokentihi"
let users = [
    {"id": 1, "firstname": "Testmann", "lastname": "TestPrivatMeldingson", "email": "jeg@finnes.ikke"},
    {"id": 2, "firstname": "Sander", "lastname": "Ulset", "email": "sander.ulset@verisure.no"}
]
let messages = []

const userApi = {
    listUsers: async () => {return users},
    getProfileData: async (userToken) => {return users[0]}
}

const messageApi = {

}


module.exports = {userData, userToken, userApi, messageApi, users}