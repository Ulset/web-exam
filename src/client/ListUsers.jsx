import React, {useState} from 'react'

const ListUsers = ({userApi, userData}) => {
    let [users, setUsers] = useState()
    const selfId = userData.id //The id of the currently logged in user.
    if(!users){
        /*If there are no loaded users, load from server*/
        userApi.listUsers().then(
            d => {
                let usersHtml = d.map(u => {
                    let {id, firstname, lastname, email} = u
                    return <li key={id}><p>{firstname} {lastname}, {email}{selfId===id ? ' (deg)' : ''}</p></li>
                })
                setUsers(usersHtml)
            }
        )
    }
    //Return statement, will tell you if you have no friends
    return <div>
        <h4>Eksisterende brukere</h4>
        <ul>{users ?
            users
            : <p>Laster..</p>}</ul>
        {users && users.length === 1 ? <p>Det er visst bare deg her! Trenger to for Ã¥ chatte! :)</p> : ''}
    </div>
}

export default ListUsers