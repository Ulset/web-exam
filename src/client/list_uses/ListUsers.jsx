import React, {useEffect, useState} from 'react'
import {InputField} from "../helpers";

const ListUsers = ({userApi}) => {
    let [users, setUsers] = useState()
    if(!users){
        userApi.listUsers().then(
            d => {
                let usersHtml = d.map(u => {
                    let {id, firstname, lastname, email} = u
                    return <div key={id}>{firstname} {lastname}, {email}</div>
                })
                setUsers(usersHtml)
            }
        )
    }

    return <div>
        <h4>Eksisterende brukere</h4>
        <div>{users ?
            users
            : <p>Laster..</p>}</div>
        {users && users.length === 1 ? <p>Det er bare en bruker her, venligst logg inn med en ekstra Google konto :)</p> : ''}

    </div>
}

export default ListUsers