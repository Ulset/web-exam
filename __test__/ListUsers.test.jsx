import {act} from "react-dom/test-utils";
import React from "react";
import ReactDOM from "react-dom";
import {userApi, userData, users} from "./testHelpers";
import ListUsers from "../src/client/ListUsers";
import {it} from "@jest/globals";

it("Lists users", async ()=>{
    const container = document.createElement("div")
    document.body.appendChild(container)
    await act(async ()=>{
        ReactDOM.render(<ListUsers userData={userData} userApi={userApi}/>, container)
    })
    for(let userEl of users){
        expect(container.innerHTML).toContain(userEl.firstname)
        expect(container.innerHTML).toContain(userEl.lastname)
        expect(container.innerHTML).toContain(userEl.email)
    }
})
it("looks the same", async ()=>{
    const container = document.createElement("div")
    document.body.appendChild(container)
    await act(async ()=>{
        ReactDOM.render(<ListUsers userData={userData} userApi={userApi}/>, container)
    })
    expect(container.innerHTML).toMatchSnapshot()
})
