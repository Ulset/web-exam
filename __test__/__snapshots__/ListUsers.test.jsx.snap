// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`looks the same 1`] = `"<div><h4>Eksisterende brukere</h4><ul><li><p>Testmann TestPrivatMeldingson, jeg@finnes.ikke</p></li><li><p>Sander Ulset, sander.ulset@verisure.no</p></li></ul></div>"`;
